### DISCUSSION TITLE
Cypress Plugin for Hostname Is Empty String

### DISCUSSION THREAD

**Discussion by tgf-coder** [cypress-io/cypress#15004] [Questions and Help] [answered]
I created a plugin in support/index.js

```
import './commands';
const os = require('os');


module.exports = (on, config) => {
  config.env.testing_computer = os.hostname();
  return config;
}
```
Basically I want to store the hostname in the environment variable testing_computer.
However when I try to access it later in a custom command by Cypress.env("testing_computer") it is an empty string. How can I get the actual value?

**Comment by amirrustam** [comment, marked as answer]
@tgf-coder you have to do this within the plugins file: https://docs.cypress.io/guides/tooling/plugins-guide.html#Use-Cases

Your current code is in the "support file" which only runs within the browser and not on your local system.

**Reply by tgf-coder** [reply]
Ok I changed it to plugins/index.js and it still has the same issue. Its an empty string

**Comment by bahmutov** [comment]
I pushed my code into https://github.com/cypress-io/cypress-test-tiny/tree/osname

```js
// cypress/plugins/index.js
const os = require('os');
module.exports = (on, config) => {
  config.env.testing_computer = os.hostname();
  console.log('config env is', config.env)
  return config;
}
```

```js
// cypress/integration/spec.js
/// <reference types="cypress" />
it('has hostname', () => {
  expect(Cypress.env('testing_computer')).to.be.a('string').and.not.be.empty
})
```

When Cypress opens you can find the hostname under Settings / configuration / env object

<img width="799" alt="Screen Shot 2021-02-09 at 2 42 37 PM" src="https://user-images.githubusercontent.com/2212006/107419050-58837f80-6ae5-11eb-87b8-e532570dfa26.png">

And in the test you can see the value

<img width="454" alt="Screen Shot 2021-02-09 at 2 43 12 PM" src="https://user-images.githubusercontent.com/2212006/107419069-5e796080-6ae5-11eb-9d29-930498c12271.png">

**Comment by tgf-coder** [comment]
Solved. I ended up clearing the contents of the plugin/index.js file and everything is now working.
